airbyte-integrations/connectors/source-stock-ticker-api/source_stock_ticker_api/source.py

```diff
@@ -0,0 +1,107 @@
+                latest_record_date = datetime.strptime(record[self.cursor_field], '%Y-%m-%d')
+                self._cursor_value = max(self._cursor_value, latest_record_date)
+            yield record
+
+    def parse_response(
+            self,
+            response: requests.Response,
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        # The response is a simple JSON whose schema matches our stream's schema exactly,
+        # so we just return a list containing the response
+        return [response.json()]
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        # The API does not offer pagination,
+        # so we return None to indicate there are no more pages in the response
+        return None
+
+    def _chunk_date_range(self, start_date: datetime) -> List[Mapping[str, Any]]:
+        """
+        Returns a list of each day between the start date and now.
+        The return value is a list of dicts {'date': date_string}.
+        """
+        dates = []
+        while start_date < datetime.now():
+            dates.append({self.cursor_field: start_date.strftime('%Y-%m-%d')})
+            start_date += timedelta(days=1)
+        return dates
+
+    def stream_slices(self, sync_mode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None) -> Iterable[Optional[Mapping[str, Any]]]:
+        start_date = datetime.strptime(stream_state[self.cursor_field], '%Y-%m-%d') if stream_state and self.cursor_field in stream_state else self.start_date
+        return self._chunk_date_range(start_date)
+
+    @property
+    def state(self) -> Mapping[str, Any]:
+        if self._cursor_value:
+            return {self.cursor_field: self._cursor_value.strftime('%Y-%m-%d')}
+        else:
+            return {self.cursor_field: self.start_date.strftime('%Y-%m-%d')}
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        self._cursor_value = datetime.strptime(value[self.cursor_field], '%Y-%m-%d')
+
+
+class SourceStockTickerApi(AbstractSource):
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        accepted_currencies = {"USD", "JPY", "BGN", "CZK", "DKK"}  # assume these are the only allowed currencies
+        input_currency = config['base']
+        if input_currency not in accepted_currencies:
+            return False, f"Input currency {input_currency} is invalid. Please input one of the following currencies: {accepted_currencies}"
+        else:
+            return True, None
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        auth = NoAuth()
+        # Parse the date from a string into a datetime object
+        start_date = datetime.strptime(config['start_date'], '%Y-%m-%d')
+        return [ExchangeRates(authenticator=auth, config=config, start_date=start_date)]
```

Comment:
- what's the sense to pass authenticator here if it is not used further?
- what's the sense to pass both `start_date` and `config` here? can't we parse `start_date` in the stream's `__init__`?